### PR TITLE
[ci-visibility] Trace command: add support for Github, Gitlab, Azure, AWS and Buildkite

### PR DESCRIPTION
**Pull request by rodrigo-roca**
### What and why?
Closes https://github.com/DataDog/datadog-ci/issues/1262 and https://github.com/DataDog/datadog-ci/issues/1169

This PR adds support for custom commands for the following providers:
* Github Actions
* Gitlab
* Azure pipelines
* AWS Codepipeline
* Buildkite

This change should be a noop for Jenkins and CircleCI.

### How?

Now the trace command sends the data to a new endpoint that can handle all of the providers mentioned above.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/trace/__tests__/trace.test.ts [hunk 1]
```diff
@@ -15,56 +31,112 @@ describe('trace', () => {
...
+        CODEBUILD_BUILD_ARN: 'arn:aws:codebuild:us-west-2:123456789012:build/MyProjectName:6a8f0d8a',
+      })
+      expect(code).toBe(0)
+      const dryRunOutput = context.stdout.toString()
+      expect(dryRunOutput).toContain('\\"DD_PIPELINE_EXECUTION_ID\\":\\"def-234\\"')
+      expect(dryRunOutput).toContain(
+        '\\"CODEBUILD_BUILD_ARN\\":\\"arn:aws:codebuild:us-west-2:123456789012:build/MyProjectName:6a8f0d8a\\"'
+      )
+    })
+    test('should detect the buidlkite environment', async () => {
```
Comment (juan-fernandez) [inline src/commands/trace/__tests__/trace.test.ts:130]:
```suggestion
    test('should detect the buildkite environment', async () => {
```

**Review by juan-fernandez — COMMENTED** [review]
I haven't finished with the review, but I think this is worth noting before continuing: 

Before the changes:

* if the CI provider was not recognized and `--no-fail` was not passed: the `execute` command threw. No span was reported.
* if the CI provider was not recognized and `--no-fail` was passed: the `execute` command did not throw and finished with the `exitCode` of the subprocess. No span was reported.

After the changes:
* if the CI provider is not recognized and `--no-fail` is not passed: the `execute` command returns exit code 1. I'm not sure if this is exactly the same as throwing. Probably yes, but worth checking the exit code of the datadog-ci command. No span was reported.
* :warning: if the CI provider is not recognized and `--no-fail` is passed: the `execute` command does not throw and returns 0, not the `exitCode` of the subprocess. No span is reported. **This worries me more**, as the exit code is changed from whatever the subprocess reported to 0

File: src/commands/trace/README.md [hunk 2]
```diff
@@ -16,14 +16,18 @@ datadog-ci trace --name "Say Hello" -- echo "Hello World"
...
+    The `value` must be a number.
 - `--no-fail` (default: `false`) will prevent the trace command from failing even when not run in a supported CI Provider. In this case, the command will be launched and nothing will be reported to Datadog.
+- `--dry-run` (default: `false`) it will run the command without sending the custom span. All other checks are performed.
 
 #### Environment variables
 
 Additionally you might configure the `trace` command with environment variables:
 
 - `DD_API_KEY` (**required**): API key used to authenticate the requests.
-- `DD_ENV`: you may choose the environment you want your test results to appear in.
```
Comment (juan-fernandez) [inline src/commands/trace/README.md:26]: did we ever read `DD_ENV` ? Are you removing it because it was a mistake?

**Review by juan-fernandez — APPROVED** [review]

File: src/commands/trace/README.md (hunk 2 above)
Comment (rodrigo-roca) [reply src/commands/trace/README.md:26]: Yes I checked the code and we never read `DD_ENV` so I just removed it from the readme 😬

File: src/commands/trace/README.md [hunk 3]
```diff
@@ -16,14 +16,18 @@ datadog-ci trace --name "Say Hello" -- echo "Hello World"
 
 - The positional arguments are the command which will be launched and traced.
 - `--name` (default: same as <command>) is a human-friendly name for the reported span.
+- `--tags` is an array of key value pairs of the shape `key:value`. This will be the tags added to the custom span.
```
Comment (maycmlee) [inline src/commands/trace/README.md:19]:
```suggestion
- `--tags` is an array of key-value pairs with the format `key:value`. These tags are added to the custom span.
```

File: src/commands/trace/README.md [hunk 4]
```diff
@@ -16,14 +16,18 @@ datadog-ci trace --name "Say Hello" -- echo "Hello World"
...
+    The resulting dictionary will be merged with whatever is in the `DD_TAGS` environment variable. If a `key` appears both in `--tags` and `DD_TAGS`, whatever value is in `DD_TAGS` will take precedence.
```
Comment (maycmlee) [inline src/commands/trace/README.md:20]:
```suggestion
    The resulting dictionary is merged with what is in the `DD_TAGS` environment variable. If a `key` appears both in `--tags` and `DD_TAGS`, the value in `DD_TAGS` takes precedence.
```

File: src/commands/trace/README.md [hunk 5]
```diff
@@ -16,14 +16,18 @@ datadog-ci trace --name "Say Hello" -- echo "Hello World"
...
+- `--measures` is an array of key value pairs of the shape `key:value`. This will be the measures added to the custom span.
```
Comment (maycmlee) [inline src/commands/trace/README.md:21]:
```suggestion
- `--measures` is an array of key-value pairs with the format `key:value`. These measures are added to the custom span.
```

File: src/commands/trace/README.md [hunk 6]
```diff
@@ -16,14 +16,18 @@ datadog-ci trace --name "Say Hello" -- echo "Hello World"
...
+    The `value` must be a number.
 - `--no-fail` (default: `false`) will prevent the trace command from failing even when not run in a supported CI Provider. In this case, the command will be launched and nothing will be reported to Datadog.
+- `--dry-run` (default: `false`) it will run the command without sending the custom span. All other checks are performed.
```
Comment (maycmlee) [inline src/commands/trace/README.md:24]:
```suggestion
- `--dry-run` (default: `false`) runs the command without sending the custom span. All other checks are performed.
```

**Review by maycmlee — COMMENTED** [review]
Left some suggestions!

**Review by maycmlee — APPROVED** [review]

**Pull request merged**